data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/http/WebClientCloudEventSender.java

```diff
@@ -130,13 +139,36 @@ private Future<HttpResponse<Buffer>> send(final CloudEvent event, final int retr
       cause -> {
         if (cause instanceof ResponseFailureException) {
           final var response = ((ResponseFailureException) cause).getResponse();
+
+          Tags tags;
+          if (response == null) {
+            tags = this.noResponseResourceTags;
+          } else {
+            tags = this.consumerVerticleContext.getTags().and(
+              Tag.of(Metrics.Tags.RESPONSE_CODE_CLASS, response.statusCode() / 100 + "xx"),
+              Tag.of(Metrics.Tags.RESPONSE_CODE, Integer.toString(response.statusCode()))
+            );
+          }
+
           if (isRetryableStatusCode(response.statusCode()) && retryCounter < consumerVerticleContext.getEgressConfig().getRetry()) {
+
+            Metrics
+            .eventCount(tags)
+            .register(meterRegistry)
+            .increment();
+      
             return retry(retryCounter, event);
           }
           return Future.failedFuture(cause);
         }
 
         if (retryCounter < consumerVerticleContext.getEgressConfig().getRetry()) {
+
+          Metrics
+          .eventCount(this.consumerVerticleContext.getTags())
+          .register(meterRegistry)
+          .increment();          
+
```

Comment:
We can do the metric increment as part of the `retry` method directly, something like this:

```java
  private Future<HttpResponse<Buffer>> retry(int retryCounter, HttpResponse response, CloudEvent event) {
    final var tags = getTags(response);  
    Metrics
      .eventDispatchRetriesCount(tags)
      .register(meterRegistry)
      .increment();

    Promise<HttpResponse<Buffer>> r = Promise.promise();
    final var delay = retryPolicyFunc.apply(retryCounter + 1);
    vertx.setTimer(delay, v -> send(event, retryCounter + 1).onComplete(r));
    return r.future();
  }
```

in this way, we remove the duplication

Reply:
addressed